### PR TITLE
pluralize #Example, remove newline after it and add note to Contributing

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

**Comment by KristofferC** [comment]
And this will of course mess up all the line numbers for the doctests........

**Comment by tkelman** [comment]
https://github.com/JuliaLang/julia/pull/22115#issuecomment-304840445

File: base/bitarray.jl [hunk 1]
```diff
@@ -1388,8 +1388,7 @@ with elements shifted `n` positions forward, filling with `false`
 values. If `n < 0`, elements are shifted backwards. Equivalent to
 `B << -n`.
 
-## Example
-
+## Examples
```
Comment (tkelman) [inline base/bitarray.jl]: single #

File: base/bitarray.jl [hunk 2]
```diff
@@ -1428,7 +1427,6 @@ values. If `n < 0`, elements are shifted forwards. Equivalent to
 `B >> -n`.
 
 ## Examples
```
Comment (tkelman) [inline base/bitarray.jl]: single #

**Comment by KristofferC** [comment]
Probably best to merge this one last and run the script on it when rebased on the other doctests PRs.

File: base/linalg/blas.jl [hunk 3]
```diff
@@ -218,7 +218,7 @@ scal(n, DA, DX, incx) = scal!(n, DA, copy(DX), incx)
 Dot product of two vectors consisting of `n` elements of array `X` with stride `incx` and
 `n` elements of array `Y` with stride `incy`.
 
-# Example:
+# Examples:
```
Comment (Sacha0) [inline base/linalg/blas.jl]: Mysterious colon?

**Review by Sacha0 — APPROVED** [review]
lgtm modulo the minor comment on blas.jl! :)

Perhaps we should codify this convention in the documentation documentation?

**Comment by KristofferC** [comment]
Rebased and updated

File: base/linalg/blas.jl (hunk 3 above)
Comment (KristofferC) [inline base/linalg/blas.jl]: Fixed this in rebased version

File: base/docs/helpdb/Base.jl [hunk 4]
```diff
@@ -1838,13 +1835,15 @@ type automatically.
...
+    # body
 
-        retval::Float64
-    end
+    retval::Float64
+end
 
-    bar = cfunction(foo, Float64, ())
+bar = cfunction(foo, Float64, ())
+``
```
Comment (tkelman) [inline base/docs/helpdb/Base.jl]: one too few backticks

File: base/array.jl [hunk 5]
```diff
@@ -420,6 +420,11 @@ julia> collect(Float64, 1:2:5)
  1.0
  3.0
  5.0
+
+julia> collect(Dict(1 => 2, 2 => "foo"))
```
Comment (tkelman) [inline base/array.jl]: did this get moved or was this always added in this pr? conflict resolution wrong way around maybe?

Comment (KristofferC) [inline base/array.jl]: Yeah, this should not be here.

Comment (tkelman) [inline base/array.jl]: should it be somewhere else? I don't see an exact match for this doctest anywhere, unless someone else deleted it?

Comment (KristofferC) [inline base/array.jl]: No, I added it wrongly.

Comment (tkelman) [inline base/array.jl]: k. would be good for a different pr (and probably different collect method's docstring) but not needed now

Comment (KristofferC) [inline base/array.jl]: yeah, ill add it later

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -182,7 +182,7 @@ Examples written within docstrings can be used as testcases known as "doctests"
     "DOCSTRING TEST"
     ```
 
-A doctest needs to match an interactive REPL including the `julia>` prompt. To run doctests you need to run `make -C doc check` from the root directory.
+A doctest needs to match an interactive REPL including the `julia>` prompt. To run doctests you need to run `make -C doc check` from the root directory. It is recommended to add the header `# Examples` above the doctests.
```
Comment (Sacha0) [inline CONTRIBUTING.md]: Perhaps specify "...above the doctests without any intervening space." or so?

**Review by Sacha0 — APPROVED** [review]
lgtm for the fixups! :)

**Comment by KristofferC** [comment]
I suggest swapping all the `# Examples` to be in bold, i.e `**Examples**`. This will make it render better in terminal and will have no effect in the generated html, ref #22870

**Comment by StefanKarpinski** [comment]
I vehemently dislike that approach – it removes useful and valid semantic information in exchange for hacked in formatting. This is like not liking how ordered lists are rendered and going through and changing all instances of ordered lists in the docs to use escaped numbers. The "Examples" section is exactly that – a section. What happens when we have another section? We also just use bold text on a line by itself for that? Why bother having a section syntax at all? Let's just use bold text on a line by itself for all of our section headers.

**Comment by KristofferC** [comment]
When it comes to things that are easy to change, I value the actual end result that users experience more than appealing to what is technically correct. We could always switch back to headers when they actually render nicely, it would take 30 seconds to create a PR. Anyway, let's leave it for now then. For consistency, should probably at least change the nicely rendering, hacked formatting of the `Summary:` and `Subfields:` sections to use actual sections instead then.

**Comment by StefanKarpinski** [comment]
I'm fine with a temporary fix as long as we track implementing a better fix in the future.
